modules/common/src/main/scala/surge/exceptions/SurgeException.scala

```diff
@@ -0,0 +1,9 @@
+package surge.exceptions
+
+sealed trait SurgeException {
+  def message: String
+
+  def toJavaException: Exception = new Exception(message)
+}
+
+case class SurgeCustomException(message: String) extends SurgeException
```

Comment:
Any way we can actually wrap the exception here instead of just the message string? Having the actual exception type + stack trace is actually really useful for apps who may reject commands with explicit exceptions that they can look for by explicit type on the asking end - for example in our example bank account we fail with an `InsufficientFundsException` if we try to withdraw too much from an account: https://github.com/UltimateSoftware/surge-samples/blob/main/examples/scala/src/main/scala/com/example/model/BankAccountCommandModel.scala#L32

Reply:
If we wrap the exception, it works but the jackson shows these kinds of warnings
`WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.fasterxml.jackson.databind.util.ClassUtil (file:/home/eldar/.cache/coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.11.4/jackson-databind-2.11.4.jar) to field java.lang.Throwable.detailMessage
WARNING: Please consider reporting this to the maintainers of com.fasterxml.jackson.databind.util.ClassUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release`